components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/FederatedAuthenticatorConfig.java

```diff
@@ -101,9 +108,17 @@ public static FederatedAuthenticatorConfig build(OMElement federatedAuthenticato
                     Property[] propertiesArr = propertiesArrList.toArray(new Property[propertiesArrList.size()]);
                     federatedAuthenticatorConfig.setProperties(propertiesArr);
                 }
+            } else if ("DefinedBy".equals(elementName)) {
+                federatedAuthenticatorConfig.setDefinedByType(DefinedByType.valueOf(element.getText()));
             }
         }
 
+        if (federatedAuthenticatorConfig.getDefinedByType() == null) {
+            federatedAuthenticatorConfig.setDefinedByType(DefinedByType.SYSTEM);
+            LOG.debug("The defined by type is not set for the {}. Hence setting default SYSTEM value.",
+                    federatedAuthenticatorConfig.getName());
+        }
+
```

Comment:
Ideally IMO, this is redundant.
We need to make sure from where we load the auth config and those points are what should be fixed

Reply:
Addressed with commit fe95d4b60aeb8557a6ac7cfe51ce2576ee94112c